azurerm/internal/services/devtestlabs/resource_arm_dev_test_global_shutdown_schedule.go

```diff
@@ -0,0 +1,324 @@
+package devtestlabs
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/devtestlabs/mgmt/2016-05-15/dtl"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDevTestLabGlobalShutdownSchedule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDevTestLabGlobalShutdownScheduleCreateUpdate,
+		Read:   resourceArmDevTestLabGlobalShutdownScheduleRead,
+		Update: resourceArmDevTestLabGlobalShutdownScheduleCreateUpdate,
+		Delete: resourceArmDevTestLabGlobalShutdownScheduleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"location": azure.SchemaLocation(),
+
+			"target_resource_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  dtl.EnableStatusEnabled,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(dtl.EnableStatusEnabled),
+					string(dtl.EnableStatusDisabled),
+				}, false),
+			},
+
+			"daily_recurrence": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"time": {
```

Comment:
Would this make more sense as a top level field that is: `daily_recurrence_time `

Reply:
I left it like this to match the existing `dev_test_schedule` resource as well as the structure of the API but I think I agree with making it a top level attribute. I left the `flatten` and `expand` functions in place and updated them to handle a simple string instead. Would appreciate feedback on that approach.